lib/utils/get-module-path-for.js

```diff
@@ -18,12 +18,32 @@ for (let packagePath of packagePaths) {
   let packageDir = path.dirname(path.resolve(cwd, packagePath));
 
   if (pkg.keywords && pkg.keywords.includes('ember-addon')) {
-    ADDON_PATHS[packageDir] = pkg.name;
+    // build addon instance
+    ADDON_PATHS[packageDir] = getAddonPackageName(packagePath);
   } else if (isEmberCliProject(pkg)) {
     APP_PATHS[packageDir] = pkg.name;
   }
 }
 
+/**
+ * takes a package path and returns the runtime name of the addon.
+ *
+ * @param {String} packagePath the path on disk (from current working directory)
+ * @returns {String} The runtime name of the addon
+ */
+function getAddonPackageName(packagePath) {
+  const indexPath = path.resolve(path.dirname(packagePath), 'index.js');
```

Comment:
We shouldn't assume `packagePath + index.js` is the right addon entry point. We have to respect `ember-addon.main` in package.json.